components/prism-yaml.js

```diff
@@ -31,7 +31,7 @@ Prism.languages.yaml = {
 		alias: 'important'
 	},
 	'string': {
-		pattern: /([:\-,[{]\s*(?:![^\s]+)?[ \t]*)("|')(?:(?!\2)[^\\\r\n]|\\.)*\2(?=[ \t]*(?:$|,|]|}))/m,
+		pattern: /([:\-,[{]\s*(?:![^\s]+)?[ \t]*)("|')(?:(?!\2)[^\\\r\n]|\\.)*\2(?=[ \t]*(?:$|,|]|}|(?:\s*#)))/m,
```

Comment:
I meant that you don't need the group at all...
Sorry for not making myself clear enough.

Reply:
My mistake, I had assumed that a grouping would be necessary. I just tried it and you are right.